shaders/mult_mat_int.fs

```diff
@@ -0,0 +1,75 @@
+#ifdef GL_FRAGMENT_PRECISION_HIGH
+    precision highp float;
+#else
+    precision mediump float;
+#endif
+
+uniform sampler2D texture0;
+uniform sampler2D texture1;
+
+varying vec2 vTexCoord;
+
+vec4 pack(float value)
+{
+    if (value == 0.0) return vec4(0);
+
+    float exponent;
+    float mantissa;
+    vec4 result;
+    float sgn;
+
+    sgn = step(0.0, -value);
+    value = abs(value);
+
+    exponent = floor(log2(value));
+    mantissa = value * pow(2.0, -exponent) - 1.0;
+    exponent = exponent + 127.0;
+    result = vec4(0);
+
+    result.a = floor(exponent / 2.0);
+    exponent = exponent - result.a * 2.0;
+    result.a = result.a + 128.0 * sgn;
+
+    result.b = floor(mantissa * 128.0);
+    mantissa = mantissa - result.b / 128.0;
+    result.b = result.b + exponent * 128.0;
+
+    result.g = floor(mantissa * 32768.0);
+    mantissa = mantissa - result.g / 32768.0;
+
+    result.r = floor(mantissa * 8388608.0);
+
+    return result / 255.0;
+}
+
+float unpack(vec4 texel)
+{
+    float exponent;
+    float mantissa;
+    float value;
+    float sgn;
+
+    sgn = -step(128.0, texel.a);
+    texel.a += 128.0 * sgn;
+
+    exponent = step(128.0, texel.b);
+    texel.b -= exponent * 128.0;
+    exponent += 2.0 * texel.a - 127.0;
+
+    mantissa = texel.b * 65536.0 + texel.g * 256.0 + texel.r;
+    value = pow(-1.0, sgn) * exp2(exponent) * (1.0 + mantissa * exp2(-23.0));
+
+    return value;
+}
+
+void main (void)
+{
+    float i = vTexCoord.s;
+    float j = vTexCoord.t;
+    float result = 0.0;
+    for (float k = 0.0; k < 4.0; ++k)
```

Comment:
why 4? why k is a `float`?

Reply:
To reduce the value of k to lie between 0 to 1, we are dividing it by 4. This is to make sure that the coordinates of the texture element we are multiplying lies in the range 0 to 1 in the texture coordinate system.
Foe texture2D function, the second parameter is vec2 which contains floats as its elements. This is the reason for k being a floating point value.

Reply:
Hi, sorry got bogged down by other work. I still do not understand why 4.0 and if 4.0 is necessary (I presume the matrix size?). If this is indeed the matrix size, it should be configurable as a uniform and not hard-coded. Please change that.

Reply:
Hi Jakub, yes its for size.
So, we are testing for a generalized notation and will update soon, can we use something like the const kspan for indicating this 4.0 as its tested for this size.